autopts/pybtp/iutctl_common.py

```diff
@@ -87,9 +89,10 @@ def read(self, timeout=20.0):
             data_memview = data_memview[nbytes:]
             toread_data_len -= nbytes
 
-        tuple_data = bytes(str(dec_data(data)), 'utf-8').decode("unicode_escape").replace("b'", "'")
+        data_decode = binascii.hexlify(data)
+        data_string = re.sub(r"(?<=\S)(?=(\S\S)+\b)", " ", data_decode.decode('utf-8'))
```

Comment:
The solution works, but has a very high computational complexity.
Please test out this example:
```python
import re
import time

test_bytes = ['0' for i in range(1000000)]
test_string = ''.join(test_bytes)

start = time.time()
result = ' '.join(f'{test_string[i:i+2]}' for i in range(0, len(test_string), 2))
end = time.time()
# Printed 0.3820197582244873 seconds
print(end - start)

start = time.time()
result = data_string = re.sub(r"(?<=\S)(?=(\S\S)+\b)", " ", test_string)
end = time.time()
# 5 minutes passed and I am still waiting for this print
print(end - start)
```
So my suggestion:
```suggestion
        data_string = binascii.hexlify(data).decode('utf-8')
        data_string = ' '.join(f'{data_string[i:i+2]}' for i in range(0, len(data_string), 2))
```